crates/solidity/inputs/language/src/definition.rs

```diff
@@ -6733,42 +6733,42 @@ codegen_language_macros::compile!(Language(
             enabled = Till("0.5.0")
         ),
         BuiltInType(
-            name = "$abiType",
+            name = "$AbiType",
             fields = [],
             functions = [
                 BuiltInFunction(
                     name = "decode",
-                    parameters = ["bytes memory", "$args"],
-                    return_type = "$args",
+                    parameters = ["bytes memory", "$Types"],
+                    return_type = "$Types",
                     enabled = From("0.5.0")
                 ),
                 BuiltInFunction(
                     name = "encode",
-                    parameters = ["$args"],
+                    parameters = ["$Types"],
```

Comment:
in order to generate a usable `built_ins.sol`, I wonder if we should add parameter names to all of these?

Reply:
I haven't done that out of laziness, but I can add the names.

This case though, is one of those that cannot be written in valid Solidity, because the `%Types` argument here is a varargs `...`. There are other similar cases.

Reply:
Aha. I think there are a few inconsistencies:

- We need a valid Solidity type, to be able to display to the users/bind in the future. Maybe we can use an array instead of the varargs, similar to how TypeScript type system does it?
- I realized that it is not consistent, since Yul parameters are untyped. Our `BuiltInFunction` type doesn't differentiate between them..
- Very few signatures have a "storage location" like `memory`. If we don't have this data, maybe we can remove the location from all of them? until we actually have a use for it, and can validate/test its impact.

Reply:
>Very few signatures have a "storage location" like memory. If we don't have this data, maybe we can remove the location from all of them? until we actually have a use for it, and can validate/test its impact.

I took this from the Solidity docs. IIUC you need storage location only for types that don't fit an EVM register. I think all instances of such parameters are annotated for the built-ins.

Reply:
>I realized that it is not consistent, since Yul parameters are untyped. Our BuiltInFunction type doesn't differentiate between them..

Hmm... good point. I didn't foresee this, but it's a problem for defining the Yul built-ins.

Reply:
>We need a valid Solidity type, to be able to display to the users/bind in the future. Maybe we can use an array instead of the varargs, similar to how TypeScript type system does it?

An array sounds good, but of what type?